js/windowMgr.js

```diff
@@ -174,6 +175,28 @@ function doCreateMainWindow(initialUrl, initialBounds) {
 
     mainWindow.on('closed', destroyAllWindows);
 
+    // Manage File Downloads
+    mainWindow.webContents.session.on('will-download', (event, item, webContents) => {
+        // When download is in progress, send necessary data to indicate the same
+        mainWindow.send('downloadProgress');
+
+        // Send file path when download is complete
+        item.once('done', (event, state) => {
+            if (state === 'completed') {
+                let data = {
+                    _id: getGuid(),
+                    filePath: item.getSavePath(),
+                    total: filesize(item.getTotalBytes()),
+                    fileName: item.getFilename(),
+                    mimeType: item.getMimeType()
+                };
+                mainWindow.send('download-completed', data);
+            } else {
+                console.log(`Download failed: ${state}`);
```

Comment:
This is a little tricky to handle because of the way downloads work on the client app. There is now way for us to know on the Electron side whether the file download has failed. We can know when it succeeds, but hard to know whether it has failed. Will discuss more on this on call. I'll remove the else though (that is probably not required).